File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result(2,0);
        unordered_map<int, int> mp; 
        for (int i = 0; i < nums.size(); ++i) {
            if (mp.contains(target - nums[i])) {
                result[0] = mp[target - nums[i]];
                result[1] = i;
            }
            mp[nums[i]] = i;
        }
        return result;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = {}
        for n in nums:
            if n in mp:
                mp[n] += 1
            else:
                mp[n] = 1
        buckets = [[] for _ in range(len(nums) + 1)]
        for key, value in mp.items():
            buckets[value].append(key)
        result = []
        for i in range(len(buckets) - 1, 0, -1):
            for num in buckets[i]:
                result.append(num)
                if len(result) == k:
                    return result
